core/shared/src/main/scala/cats/effect/IO.scala

```
@@ -821,8 +821,17 @@ sealed abstract class IO[+A] private () extends IOPlatform[A] {
    * Note that errors still get logged (via IO's internal logger), because errors being thrown
    * should never be totally silent.
    */
-  def unsafeRunAndForget()(implicit runtime: unsafe.IORuntime): Unit =
-    unsafeRunAsync(_ => ())
+  def unsafeRunAndForget()(implicit runtime: unsafe.IORuntime): Unit = {
+    val _ = unsafeRunFiber((), _ => (), _ => ())
```

Since this is no longer calling delegating to `unsafeRunAsync`, that means it won't throw a `new CancellationException("The fiber was canceled")` when cancelled, correct? Just checking that's intended?

---

Good catch! That regression was *not* intended.

---

Oh actually, looking at this more, the cancelation exception was eaten *anyway*, so this is still matching the existing behavior.

---

Gotcha, thanks for explaining!